tik/lib/include/tik/Kernel.h

```diff
@@ -22,21 +22,38 @@ namespace TraceAtlas::tik
         std::string GetHeaderDeclaration(std::set<llvm::StructType *> &AllStructures);
         std::string Name;
         std::set<llvm::BasicBlock *> Conditional;
+        struct KICompare
+        {
+            using is_transparent = void;
+            bool operator()(const std::shared_ptr<KernelInterface> &lhs, const std::shared_ptr<KernelInterface> &rhs) const
+            {
+                return lhs->Block < rhs->Block;
+            }
+            bool operator()(const std::shared_ptr<KernelInterface> &lhs, int64_t rhs) const
+            {
+                return lhs->Block < rhs;
+            }
+            bool operator()(int64_t lhs, const std::shared_ptr<KernelInterface> &rhs) const
+            {
+                return lhs < rhs->Block;
+            }
+        };
```

Comment:
I'm confused. Why is this necessary? Can this not just be built into the KernelInterface?

I don't understand what is happening here.

Reply:
This allows me to search the sets Entrances, Exits that have KernelInterface objects with block IDs only, not KernelInterface objects

Reply:
I'm going to ask for a more distinct name or a comment explaining that then.